extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -0,0 +1,232 @@
+      };
+      req.send();
+    }.bind(this));
+  }
+
+  constructUrlFromMedia_(id) {
+    return this.rendererBaseUrl_ + '/interaction/' + id;
+  }
+
+  constructIframe_(src) {
+    const iframe = this.iframe_;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('scrolling', 'no');
+    iframe.src = src;
+    iframe.height = this.element.getAttribute('height');
+    iframe.classList.add('-amp-apester-iframe');
+    this.applyFillContent(iframe);
+    return iframe;
+  }
+
+  //TODO remove overflow when 'changeHeight' PR approved
+  constructOverflow_() {
+    const overflow = this.element.ownerDocument.createElement('div');
+    overflow.setAttribute('overflow', '');
+    overflow.className = '-amp-apester-overflow-container';
+    const overflowButton = this.element.ownerDocument.createElement('button');
+    overflowButton.className = '-amp-apester-overflow-button';
+    overflowButton.innerHTML = 'Start Here';
+    overflow.appendChild(overflowButton);
+    return overflow;
+  }
+
+  /** @override */
+  layoutCallback() {
+    return this.queryMedia_()
+      .then(media => {
+        const src = this.constructUrlFromMedia_(media.interactionId);
+        const iframe = this.constructIframe_(src);
+        const overflow = this.constructOverflow_();
+        const mutate = state => {
+          state.element.classList.add('-amp-apester-iframe-ready');
+        };
+        const state = {
+          element: iframe, mutator: mutate,
+        };
+        this.iframe_ = iframe;
+        this.element.appendChild(overflow);
+        this.element.appendChild(iframe);
+        return this.iframePromise_ = loadPromise(iframe).then(() => {
+          vsyncFor(this.win).runPromise({mutate}, state);
+          return media;
+        });
+      }, () => {
+        return undefined;
+      }).then(media => {
+        // hide placeholder
+        this.togglePlaceholder(false);
+        let height = 0;
+        if (media) {
+          height = media.data.size.height;
+          const width = media.data.size.width;
+          height += (media.layout.directive === 'contest-poll') ? 40 : 0;
+          const amp = this.element;
+          amp.setAttribute('height', height);
+          amp.setAttribute('width', width);
+        }
+
+        //TODO: PR change from attemptChangeHeight to changeHeight
+        this./*OK*/attemptChangeHeight(height);
+      });
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const img = this.element.ownerDocument.createElement('amp-img');
+    const placeholder = this.element.ownerDocument.createElement('div');
+    placeholder.setAttribute('placeholder', '');
+    placeholder.height = this.element.getAttribute('height');
+    placeholder.className = '-amp-apester-loader-container';
+    img.className = '-amp-apester-loader';
+    img.setAttribute('src',
+                     'https://images.apester.com/images%2Floader.gif');
```

Comment:
@dvoytenko is it ok to load this GIF using `amp-img`? I don't think `amp-anim` would work here since it won't be prerendered.

I think this is ok just want to make sure.